pkg/sql/opt/memo/extract.go

```diff
@@ -540,3 +543,126 @@ func ExtractTailCalls(expr opt.Expr, tailCalls map[opt.ScalarExpr]struct{}) {
 		}
 	}
 }
+
+// ExtractExactScalarExprsForColumns extracts scalar expressions that tightly
+// constrain the given columns from the given filters. Unlike constraint
+// builder, this function:
+//   - constrains to scalar expressions instead of constants
+//   - always constrains tightly
+//   - in fact, always constrains _exactly_ (that is, to scalar expressions that
+//     produce not only equivalent values, but _exactly_ the same bytes)
+//
+// If a filter does not exactly constrain one of the given columns, or
+// references more than one of the given columns, the filter will be added to
+// remainingFilters. ExtractExactScalarExprsForColumns returns the mapping from
+// constrained columns to scalar expressions, the remaining filters, and a
+// boolean indicating whether all columns were constrained.
```

Comment:
nit: It may be worth noting that the returned remaining filters can include some synthesized filters.

Reply:
Bump. Also just a comment, so not blocking.